compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -472,7 +472,9 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             .predicates
             .into_iter()
         {
-            if let ty::PredicateKind::Trait(..) = super_trait.kind().skip_binder() {
+            if let ty::PredicateKind::Trait(..) | ty::PredicateKind::Projection(..) =
```

Comment:
> I actually think we want to just include all predicates here -- did you try that?

Yeah, so this was originally limited to just trait predicates in e42c97919. The test case added in that diff demonstrates why the code doesn't register outlives obligations, though upon second glance, it seems a bit suspicious.

> It might be better to check them as part of the "WF" conditions for dyn types. But I think this is an ok place to put them for now.

I agree with this, though I recall reading somewhere that there were difficulties adding these checks in WF and that's why they live here. Let me see if I can find that.